tap_shopify/__init__.py

```diff
@@ -20,8 +20,17 @@
 def initialize_shopify_client():
     api_key = Context.config['api_key']
     shop = Context.config['shop']
-    session = shopify.Session(shop, api_key)
-    shopify.ShopifyResource.activate_session(session)
+
+    if 'password' in Context.config:
+        password = Context.config['password']
+        shop_url = 'https://{}:{}@{}.myshopify.com/admin'.format(
+            api_key,
+            password,
+            shop)
+        shopify.ShopifyResource.set_site(shop_url)
+    else:
+        session = shopify.Session(shop, api_key)
+        shopify.ShopifyResource.activate_session(session)
```

Comment:
There should be some level of documented mutual exclusivity in the config styles, as well as a clear order of preference, and a log indicating which one was chosen.